spec/facility_spec.rb

```diff
@@ -16,11 +24,125 @@
 
   describe '#add service' do
     it 'can add available services' do
+      @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
       expect(@facility.services).to eq([])
       @facility.add_service('New Drivers License')
-      @facility.add_service('Renew Drivers License')
+      @facility.add_service('Renew License')
+      @facility.add_service('Vehicle Registration')
+      @facility.add_service('Written Test')
+      @facility.add_service('Road Test')
+      expect(@facility.services).to eq(['New Drivers License', 'Renew License', 'Vehicle Registration', 'Written Test', 'Road Test'])
+    end
+  end
+
+  describe '#register vehicle' do
+    it 'starts with an empty array' do
+      @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      expect(@facility.registered_vehicles).to eq([])
+    end
+
+    it 'can add vehicles to registered vehicles' do
+      @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      @facility.add_service('Vehicle Registration')
+      @facility.register_vehicle(@cruz)
+      @facility.register_vehicle(@bolt)
+      @facility.register_vehicle(@camaro)
+
+      expect(@facility.registered_vehicles).to eq([@cruz, @bolt, @camaro])
+    end
+
+    it 'sets todays date as registration date' do
+      @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      @facility.register_vehicle(@cruz)
+
+      expect(@cruz.set_registration_date).to eq(Date.today)
```

Comment:
In this assertion, we don't want to call `@cruz.set_registration_date` because that is actually running that method, which means we're not testing the behavior of the `register_vehicle` method. We'd want to change this line to `expect(@cruz.registration_date).to eq(Date.today)` which would prove that it was the `register_vehicle` method that changed the registration date for the vehicle.